src/py/pyodide/http.py

```diff
@@ -40,6 +40,17 @@ def __init__(self, status: int, status_text: str, url: str) -> None:
             super().__init__(f"{status} Client Error: {status_text} for url: {url}")
         elif 500 <= status < 600:
             super().__init__(f"{status} Server Error: {status_text} for url: {url}")
+        else:
+            super().__init__(
+                f"{status} Invalid error code not between 400 and 599: {status_text} for url: {url}"
+            )
+
+    def __reduce__(self):
+        return (
+            self.__class__,
+            (self.status, self.status_text, self.url),
+            self.__dict__,
+        )
```

Comment:
But without this the test added in test_pyodide_http.py that raises HttpError fails...

Reply:
Only when the test was raising an uncaught HttpError, so would fail for other reasons. When the test is correct, it wouldn't happen.

But how about let's just add a separate test to this PR that does `pickle.load(pickle.dumps(HttpError(...)))`. It can even run in the host Python (no need for `@run_in_pyodide`).